build/patches/Add-a-proxy-configuration-page-fix-377-679.patch

```diff
@@ -0,0 +1,236 @@
+@@ -338,8 +350,11 @@ void ProxyConfigMessageHandler::OnApply(const base::ListValue* list) {
+       proxyConfig.proxy_rules().proxies_for_ftp.SetFromPacString(ftp->GetString());
+       proxyConfig.proxy_rules().fallback_proxies.SetFromPacString(fallback->GetString());
+       readBypass = true;
+-  } else if (t == "empty") {
++  } else if (t == "direct") {
+       proxyConfig.proxy_rules().type = net::ProxyConfig::ProxyRules::Type::EMPTY;
++  } else if (t == "none") {
++      OnClear(nullptr);
++      return;
+   } else {
+      // invalid type
+      LOG(WARNING) << "invalid proxy configuration type";
+@@ -367,20 +382,23 @@ void ProxyConfigMessageHandler::apply(const net::ProxyConfig& proxyConfig) {
+   if (proxyConfig.auto_detect()) {
+     const base::Value cfg = ProxyConfigDictionary::CreateAutoDetect();
+     profile_->GetPrefs()->Set(proxy_config::prefs::kProxy, cfg);
+-    return;
+   } else if (proxyConfig.has_pac_url()) {
+     const base::Value cfg = ProxyConfigDictionary::CreatePacScript(proxyConfig.pac_url().spec(), proxyConfig.pac_mandatory());
+     profile_->GetPrefs()->Set(proxy_config::prefs::kProxy, cfg);
+-    return;
+-  }
+-
+-  auto proxyRulesAsString = proxyConfig.proxy_rules().ToString();
+-  auto bypassRulesAsString = proxyConfig.proxy_rules().bypass_rules.ToString();
++  } else if (proxyConfig.proxy_rules().type == net::ProxyConfig::ProxyRules::Type::EMPTY) {
++    const base::Value cfg = ProxyConfigDictionary::CreateDirect();
++    profile_->GetPrefs()->Set(proxy_config::prefs::kProxy, cfg);
++  } else {
++    auto proxyRulesAsString = proxyConfig.proxy_rules().ToString();
++    auto bypassRulesAsString = proxyConfig.proxy_rules().bypass_rules.ToString();
+ 
+-  // fixed servers
+-  const base::Value cfg = ProxyConfigDictionary::CreateFixedServers(proxyRulesAsString,
+-                          bypassRulesAsString, proxyConfig.proxy_rules().reverse_bypass);
+-  profile_->GetPrefs()->Set(proxy_config::prefs::kProxy, cfg);
++    // fixed servers
++    const base::Value cfg = ProxyConfigDictionary::CreateFixedServers(proxyRulesAsString,
++                            bypassRulesAsString, proxyConfig.proxy_rules().reverse_bypass);
++    profile_->GetPrefs()->Set(proxy_config::prefs::kProxy, cfg);
++  }
++  profile_->GetPrefs()->CommitPendingWrite();
++  OnEnableNotifyUIWithState(nullptr);
+ }
+ 
+ }  // namespace
+diff --git a/net/proxy_resolution/proxy_config.cc b/net/proxy_resolution/proxy_config.cc
+old mode 100644
+new mode 100755
+index 133ff6c46cfcb53938671c6ec4a22b663d00bd29..82cda2455b15a550f805d5620613185b075c5f1d
+--- a/net/proxy_resolution/proxy_config.cc
++++ b/net/proxy_resolution/proxy_config.cc
+@@ -110,7 +110,8 @@ void ProxyConfig::ProxyRules::ParseFromString(const std::string& proxy_rules) {
+                                    &single_proxies,
+                                    ProxyServer::SCHEME_HTTP);
+         type = Type::PROXY_LIST;
+-        return;
```

Comment:
I think this was fine, the bug was fixed below in the other change for the single proxy list.

Reply:
sorry I do not understand

Reply:
The bug is already fixed with the other change. I think there's no upstream bug, see the tests here: https://source.chromium.org/chromium/chromium/src/+/master:net/proxy_resolution/proxy_config_unittest.cc

Reply:
I don't see "myproxy:80 **; (semicolon)** https://myotherproxy"

so we save and upstream demands it

Reply:
I will merge it and test myself.

Reply:
you can see with

`cat /data/user/0/xxxxxxx/app_chrome/Default/Preferences | grep proxy`

where xxxxxxx is your local app, for my build are is org.chromium.chrome.stable

Reply:
I verified that it is necessary; you could send a patch upstream about this.